docs/FAQ.md

```diff
@@ -0,0 +1,19 @@
+# FAQ for s2n-tls
+
+### Why is the ClientHello version reported as TLS 1.2 but the handshake itself negotiated TLS 1.3?
+The TLS 1.3 spec uses the TLS 1.2 version in ClientHello and on record layers for compatibility with middleboxes and poor server implementations. There is a separate ClientHello extension which lists the actual versions supported.
+
+### Why is my TLS handshake failing on validating my peer’s certificate? I gave my s2n endpoint a root certificate that verifies the peer’s certificate.
+You may need to validate the peer’s hostname by implementing the s2n_verify_host_fn callback. The default behavior may need to be altered for your usecase.
```

Comment:
Can we spell out what the default behavior is here? I'm left wondering if the default behavior is to reject all.

Reply:
I'd rather we link to the usage guide section that describes that behavior. We should probably link to sections in the usage guide for most of these questions.

Reply:
yeah I am trying to avoid just copying the Usage Guide here.